lib/Resource.js

```
@@ -77,18 +77,18 @@ var Resource = function (_strategy, _config) {
     };
 
     return request(options)
-            .then(function (body) {
-              if (callback) {
-                callback(null, body);
-              } else {
-                return body;
-              }
-            })
-            .catch(function (err) {
-              if (callback) {
-                callback(err);
-              } else { return err; }
-            });
+      .then(function (body) {
+        if (callback) {
+          callback(null, body);
+        } else {
+          return body;
+        }
+      })
+      .catch(function (err) {
+        if (callback) {
+          callback(err);
+        } else { throw err; }
```

Here is the bug fix; the error needs to be thrown instead of returned.